app/src/main/java/com/example/frogweather/ui/NetworkViewModel.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.frogweather.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.frogweather.data.NetworkRepository
+import com.example.frogweather.data.Forecast
+import kotlinx.coroutines.launch
+
+class NetworkViewModel : ViewModel() {
+    private val networkRepository = NetworkRepository()
+
+    fun getForecast(latitude: Double, longitude: Double, millis: Long): LiveData<Forecast> {
+        val forecast = MutableLiveData<Forecast>()
+        viewModelScope.launch { forecast.value = networkRepository.getForecast(latitude, longitude, millis) }
```

Comment:
I think here you will have a crash when noInternet connection, please check

Reply:
yes i have  a crash. it is done